localstack/services/kinesis/kinesis_starter.py

```diff
@@ -1,214 +1,87 @@
-        "%s/node_modules/kinesalite/cli.js --shardLimit %s --port %s"
-        " --createStreamMs %s --deleteStreamMs %s --updateStreamMs %s %s"
-    ) % (
-        MODULE_MAIN_PATH,
-        config.KINESIS_SHARD_LIMIT,
-        backend_port,
-        latency,
-        latency,
-        latency,
-        kinesis_data_dir_param,
-    )
-
-    return _run_proxy_and_command(
-        cmd=cmd,
-        port=port,
-        backend_port=backend_port,
+    _server.start()
+    log_startup_message("Kinesis")
+    port = port or config.PORT_KINESIS
+    start_proxy_for_service(
+        "kinesis",
+        port,
+        backend_port=_server.port,
         update_listener=update_listener,
-        asynchronous=asynchronous,
     )
+    return _server
 
 
 def check_kinesis(expect_shutdown=False, print_error=False):
-    if expect_shutdown is False and kinesis_stopped.is_set():
-        raise AssertionError("kinesis backend has stopped")
-
     out = None
+    if not expect_shutdown:
+        assert _server
+
     try:
-        # check Kinesis
-        wait_for_port_open(PORT_KINESIS_BACKEND, http_path="/", expect_success=False, sleep_time=1)
-        endpoint_url = f"http://127.0.0.1:{PORT_KINESIS_BACKEND}"
+        _server.wait_is_up()
         out = aws_stack.connect_to_service(
-            service_name="kinesis", endpoint_url=endpoint_url
+            service_name="kinesis", endpoint_url=_server.url
         ).list_streams()
     except Exception:
         if print_error:
             LOG.exception("Kinesis health check failed")
-
     if expect_shutdown:
-        assert out is None or kinesis_stopped.is_set()
+        assert out is None
     else:
-        assert not kinesis_stopped.is_set()
-        assert out and isinstance(out.get("StreamNames"), list)
+        assert out is not None and isinstance(out.get("StreamNames"), list)
 
 
-def kinesis_running():
-    return PROCESS_THREAD is not None
+def kinesis_running() -> bool:
+    """
+    Checks if there is a currently running Kinesis server instance.
+    Currently used by localstack_ext/utils/cloud_pods.py
+    :returns: True is there is a running Kinesis server instance, False otherwise
+    TODO: rename this function to is_kinesis_running() for clarity
+    """
+    global _server
+    if _server is None:
+        return False
+    return _server.is_running()
 
 
 def restart_kinesis():
-    if PROCESS_THREAD:
-        LOG.debug("Restarting Kinesis process ...")
-        PROCESS_THREAD.stop()
-        kinesis_stopped.wait()
-        kinesis_stopped.clear()
-        start_kinesis(asynchronous=True, update_listener=kinesis_listener.UPDATE_KINESIS)
-        # giving the process some time to startup; TODO: to be replaced with service lifecycle plugin
-        time.sleep(1)
+    global _server
+    if _server:
+        _server.shutdown()
+        _server.join(timeout=10)
+        _server = None
+
+    LOG.debug("Restarting Kinesis process ...")
+    start_kinesis(update_listener=kinesis_listener)
```

Comment:
@SwatsonCodes Looks like this change was breaking persistence tests in -ext - should probably be this instead:
```
     start_kinesis(update_listener=kinesis_listener.UPDATE_KINESIS)
```
?